src/olympia/versions/models.py

```diff
@@ -175,8 +175,11 @@ def should_have_due_date(self, negate=False):
         due date instead."""
         method = getattr(self, 'exclude' if negate else 'filter')
         is_theme = Q(addon__type__in=amo.GROUP_TYPE_THEME)
+        # Note: once the auto_approval_delayed_until date is past, we don't
+        # remove the due date unless something else causes the add-on to
+        # trigger that check.
         has_auto_approval_delayed = Q(
-            addon__reviewerflags__auto_approval_delayed_until__isnull=False
+            addon__reviewerflags__auto_approval_delayed_until__gt=datetime.now()
```

Comment:
To handle versions submitted after the auto-approval delay set on the add-on has passed. Those should be auto-approved, the delay is now meaningless.

Reply:
I mean after the deadline has passed there will still be a `due_date` set unless something triggers `reset_date` to call this function, and then clear it. But either that "something" is the auto_approved command and it'll clear the `auto_approval_delayed_until` flag too (meaning the change isn't useful); or it's some other circumstance that I can't think of.

Reply:
Yes, that's fine. There will be a 5 minutes window of time where the version would be seen in the queue despite being auto-approvable, and after that `auto_approve` would remove the `due_date` when approving.

@wagnerand even mentioned that once we rework the way `needs_human_review` works, the due date should even stick in this case (since it's been set by a scanner action that set the delay + flagged the version for a reviewer to look at), despite the auto-approval.

Reply:
Reverted in https://github.com/mozilla/addons-server/pull/20181/commits/212d46d219cd42b6289fa3c206986a5e53cb9db0 following discussion, because we couldn't find a scenario where it would make a difference with the current code.